src/lib/response.ts

```diff
@@ -0,0 +1,47 @@
+export const responseMetadataKey = Symbol('response');
+const loopback = require('loopback');
+
+export abstract class ModelResponseInstance {
+  toJSON?: () => {
+    [key: string]: any
+  };
+  [key: string]: any;
+}
+
+export interface IResponseConfig { responseClass: string, isMulti?: boolean }
+
+export function Response(config: string|IResponseConfig|string[]) {
+  return function(target: any, propertyKey: string, descriptor: PropertyDescriptor) {
+    if (typeof config === 'string') {
+      config = {responseClass: config, isMulti: false};
+    } else if (Array.isArray(config)) {
+      config = {responseClass: config.pop(), isMulti: true};
+    }
+    Reflect.defineMetadata(responseMetadataKey, config, target.constructor, propertyKey);
+  }
+}
+
+export function buildResponse(result: any, RemoteClass: any) {
+  const meta: IResponseConfig =
+      Reflect.getOwnMetadata(responseMetadataKey, RemoteClass, 'onRemote');
+  if (!meta) {
+    return result;
+  }
+
+  const {isMulti, responseClass} = meta;
+  let ResponseClass;
+  ResponseClass = loopback.registry.modelBuilder.models[responseClass];
+  if (!ResponseClass) {
+    throw new Error(`No model definition for ${responseClass} found on the registry`);
```

Comment:
I think in the future we need a prod mode vs dev mode. In prod this should be fine and just pass back the payload ala lb. In dev it should be an error.